### DISCUSSION TITLE
Enable Full-Text Search by Default

### DISCUSSION THREAD

**Discussion by iansinnott** [rhashimoto/wa-sqlite#79] [Ideas]
This is just a suggestion, since there are tradeoffs.

The good:

- Parity with `sqlite3`. Since FTS 3, 4, 5 are enabled in the default build of sqlite most users are likely familiar with.
- Users don't need to figure out how to modify the `Makefile` in order to use FTS features (i.e. via questions like #77)

The bad:

- Increased bundle size. On my own system the async wasm bundle went from 1.3mb to 1.5mb. 

What would need to be done:

In the Makefile, add new `WASQLITE_DEFINES`:

```
	-DSQLITE_ENABLE_FTS5 \
	-DSQLITE_ENABLE_FTS4 \ # optional
	-DSQLITE_ENABLE_FTS3 \ # optional
```

**Comment by rhashimoto** [comment]
Another argument for:

* The official SQLite WASM build includes FTS5, by my reading of the [makefile](https://github.com/sqlite/sqlite/blob/a4d04c59baf858ea4daf14317587808996d75c69/ext/wasm/GNUmakefile#L130).

And against:

* The current build settings are based on the [recommended options](https://sqlite.org/compile.html#recommended_compile_time_options), which don't include FTS.

Unfortunately, I don't have any real data on how developers use (or want to use) wa-sqlite, so it's hard to evaluate that trade off and I should probably default to not making things bigger and slower for the existing user base. Perhaps the way to go with this would be to include two .wasm builds (each with and without Asyncify), one with the current settings and an additional "kitchen sink" variation that favors features over size and more thorough testing.

**Comment by sgbeal** [comment]
> The official SQLite WASM build includes FTS5, by my reading of the [makefile](https://github.com/sqlite/sqlite/blob/a4d04c59baf858ea4daf14317587808996d75c69/ext/wasm/GNUmakefile#L130).

Just to confirm: that is indeed the case. For the wasm build, since it's a new target platform for us, we've opted against including deprecated features by default (which includes, for example, the "double-quoted-string" support).

**Comment by rhashimoto** [comment]
I added a dist-xl build target with additional [features enabled](https://github.com/rhashimoto/wa-sqlite/blob/90134b5b99fca8818f96cfb2746b5c081ee8602e/Makefile#L98-L106), including FTS5. This target is not built (or tested) by default - instead I'll attach the artifacts produced to the [release message](https://github.com/rhashimoto/wa-sqlite/releases).
